primefaces-showcase/src/main/webapp/ui/csv/complex.xhtml

```diff
@@ -12,18 +12,22 @@
             PrimeFaces.validator['MyComplexValidator'] = {
 
                 validate: function (element) {
-                    var nameRequired = element.find('#form\\:nameRequired');
-                    var name = element.find('#form\\:name');
+                    // element is the button that triggered the validation
+                    // let nameRequiredJq = element.parent('.ui-outputpanel').find('#form\\:nameRequired');
+                    // let nameJq = element.parent('.ui-outputpanel').find('#form\\:name');
 
-                    if (nameRequired.find('.ui-chkbox-icon').hasClass('ui-icon-check') && name.val().length === 0) {
-                        PrimeFaces.validator.Highlighter.types['default'].highlight(name);
+                    /*
+                     * Instead of wrestling with jQuery parent/find-methods we could also use widgetVar-mechanism (eg PF('nameRequired') and PF('name'))
+                     * for straight-forward scenarios without e.g. some kind of ui:repeat, within p:dataTable or similar.
+                     */
+                    let nameRequiredPf = PF('nameRequired');
+                    let namePf = PF('name');
 
-                        throw {
-                            summary: 'Validation Error',
-                            detail: 'A name is required!'
-                        };
+                    if (nameRequiredPf.isChecked() && namePf.jq.val().length === 0) {
+                        PrimeFaces.validator.Highlighter.types['default'].highlight(namePf.jq);
+                        PrimeFaces.validation.ValidationContext.addMessage(namePf.jq, { summary: 'Validation Error', detail: 'A name is required!' });
```

Comment:
it it "better" / more intuitiv for users to call addMessage manually or was it broken?

Reply:
built-in validators uses:
`throw vc.getMessage(this.MINIMUM_MESSAGE_ID, min, vc.getLabel(element));`

which actually is:

```
            throw {
                summary: summary,
                detail: detail
            };
```

Reply:
```javascript
                        throw {
                            summary: 'Validation Error',
                            detail: 'A name is required!'
                        };
```

Is (for this usecase) IMO broken. At least i could not get it working.